semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java

```diff
@@ -155,4 +169,31 @@
     }
     return Optional.empty();
   }
+
+  public static Optional<AttributeValue> getValidHttpUrl(Event event) {
+    for (String key : FULL_URL_ATTRIBUTES) {
+      AttributeValue value = SpanAttributeUtils.getAttributeValue(event, key);
+      if (value != null && isValidUrl(value.getValue())) {
+        return Optional.of(value);
+      }
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * accepts any absolute or relative URL. e.g. absolute URL:
+   * http://hypertrace.org/customer?customer=392 relative URL: /customer?customer=392
+   */
+  public static boolean isValidUrl(String url) {
```

Comment:
I don't think these methods `isValidUrl` and `getNormalizedUrl` should be public. Can you please convert them to private?